.github/workflows/ci-test.yml

```diff
@@ -0,0 +1,212 @@
+        with:
+          pattern: '**.rs\|Cargo.toml\|Cargo.lock'
+      - id: non-rust-lint-changes
+        name: find shell/dockerfile changes
+        uses: ./.github/actions/matches
+        with:
+          pattern: '**.Dockerfile\|**.sh'
+      - id: dev-setup-sh-changes
+        name: find dev-setup.sh/base docker image changes
+        uses: ./.github/actions/matches
+        with:
+          pattern: '.github/workflows/ci-publish-base-image.yml\|docker/ci/**/Dockerfile\|scripts/dev_setup.sh'
+      - id: website-changes
+        name: find website changes.
+        uses: ./.github/actions/matches
+        with:
+          pattern: "documentation/**"
+
+  dev-setup-sh-test:
+    runs-on: ubuntu-latest-xl
+    continue-on-error: false
+    needs: prepare
+    if: ${{ needs.prepare.outputs.test-dev-setup == 'true' }}
+    strategy:
+      matrix:
+        target_os: [alpine, arch, centos, github]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: build image with dev-setup.sh
+        run: docker build -f docker/ci/${{ matrix.target_os }}/Dockerfile -t libra/build_environment:test .
+
+  non-rust-lint:
+    runs-on: ubuntu-latest
+    continue-on-error: false
+    needs: prepare
+    if: ${{ needs.prepare.outputs.test-non-rust-lint == 'true'  }}
+    container:
+      image: libra/build_environment:github-1
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: shell lints
+        run: |
+          shellcheck scripts/dev_setup.sh && \
+          shellcheck scripts/dockerhub_prune.sh && \
+          shellcheck .github/actions/changes/get_pr_info.sh && \
+          shellcheck docker/build_push.sh && \
+          shellcheck docker/dockerhub_to_novi_ecr.sh
+      - name: docker lints
+        run: |
+          hadolint docker/ci/github/Dockerfile && \
+          hadolint docker/ci/arch/Dockerfile && \
+          hadolint docker/ci/alpine/Dockerfile && \
+          hadolint docker/ci/centos/Dockerfile
+
+  lint:
+    runs-on: ubuntu-latest-xl
+    timeout-minutes: 30
+    continue-on-error: false
+    needs: prepare
+    container:
+      image: libra/build_environment:github-1
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: ./.github/actions/build-setup
+      - name: cargo lint
+        run: cargo x lint
+      - name: cargo clippy
+        run: cargo xclippy --workspace --all-targets
+      - name: cargo clippy tcb
+        run: cargo xclippy --members lec --members lsr --members key-manager
+      - name: cargo fmt
+        run: cargo xfmt --check
+      - uses: ./.github/actions/build-teardown
+
+  unit-test:
+    runs-on: ubuntu-latest-xl
+    timeout-minutes: 70
+    continue-on-error: false
+    needs: prepare
+    #if: ${{ needs.prepare.outputs.test-rust == 'true' }}
```

Comment:
should probably uncomment this now. and add **.move to it's triggers.

Reply:
Since that was logic you built I'm not going to mess with it right now, I'll have you build ontop once this lands to make that change (and uncomment that line)